From a0dbdec6d63231ee099483848e10ef7d0c761306 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 24 Jun 2025 13:49:20 +0800
Subject: [PATCH 29/63] loonggpu: Drop lg_ttm_tt_bind

Get rid of the ttm_tt_populate usage: this symbol is only exported if
DRM_TTM_KUNIT_TEST which requires COMPILE_TEST on LoongArch.

I plan to drop all the conftest mess and use multiple branches for
different kernel versions later.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 conftest.sh                   | 15 ---------------
 include/loonggpu_ttm_helper.h | 21 ---------------------
 loonggpu/loonggpu.Kbuild      |  1 -
 3 files changed, 37 deletions(-)

diff --git a/conftest.sh b/conftest.sh
index 12c7089..8cd20ea 100644
--- a/conftest.sh
+++ b/conftest.sh
@@ -1698,21 +1698,6 @@ compile_test() {
             compile_check_conftest "$CODE" "LG_TTM_BO_PIPELINE_MOVE" "" "functions"
         ;;
 
-        ttm_tt_bind)
-            #
-            # Determine if the ttm_tt_bind function exists.
-            #
-            # Changed by commit 9e9a153bdf2555a931fd37678a8e44d170a5d943 ("drm/ttm:
-            # move ttm binding/unbinding out of ttm_tt paths") in v5.9-rc6
-            #
-            CODE="
-            #include <drm/ttm/ttm_tt.h>
-            int conftest_ttm_tt_bind(void) {
-                return ttm_tt_bind();
-            }"
-            compile_check_conftest "$CODE" "LG_TTM_TT_BIND" "" "functions"
-        ;;
-
         ttm_bo_mem_put)
             #
             # Determine if the ttm_bo_mem_put function exists.
diff --git a/include/loonggpu_ttm_helper.h b/include/loonggpu_ttm_helper.h
index 1b082fe..b74f47d 100644
--- a/include/loonggpu_ttm_helper.h
+++ b/include/loonggpu_ttm_helper.h
@@ -152,27 +152,6 @@ static inline int lg_ttm_bo_pipeline_move(struct ttm_buffer_object *bo,
 	return r;
 }
 
-static inline int lg_ttm_tt_bind(struct ttm_buffer_object *bo, lg_ttm_mem_t *mem,
-				 struct ttm_operation_ctx *ctx)
-{
-#if defined(LG_TTM_TT_BIND)
-	return ttm_tt_bind(bo->ttm, mem, ctx);
-#else
-	int r;
-#if defined(LG_TTM_BO_POPULATE)
-	r = ttm_bo_populate(bo, ctx);
-#else
-	r = ttm_tt_populate(bo->bdev, bo->ttm, ctx);
-#endif
-	if (r)
-		return r;
-	r = loonggpu_ttm_backend_bind(bo->bdev, bo->ttm, mem);
-	if (r)
-		return r;
-	return r;
-#endif
-}
-
 static inline void lg_ttm_bo_mem_put(struct ttm_buffer_object *bo, lg_ttm_mem_t *mem)
 {
 #if defined(LG_TTM_BO_MEM_PUT) && defined(LG_DRM_TTM_TTM_BO_DRIVER_H_PRESENT)
diff --git a/loonggpu/loonggpu.Kbuild b/loonggpu/loonggpu.Kbuild
index 3fed4a5..b86419e 100644
--- a/loonggpu/loonggpu.Kbuild
+++ b/loonggpu/loonggpu.Kbuild
@@ -77,7 +77,6 @@ LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_bo_init_mm
 LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_bo_clean_mm
 LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_resource_alloc
 LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_bo_pipeline_move
-LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_tt_bind
 LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_bo_mem_put
 LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_tt_set_populated
 LG_CONFTEST_FUNCTION_COMPILE_TESTS += ttm_resource_manager_evict_all
-- 
2.52.0

